[PPM-291] Remove client statistics 11k messages

VladyslavTupikin opened the pull request:
Need to remove next messages and all related source code:

```
ACTION_CONTROL_CLIENT_STATISTICS_11K_REQUEST
ACTION_CONTROL_CLIENT_STATISTICS_11K_RESPONSE
```

https://jira.prplfoundation.org/browse/PPM-291

Commit:
common: tlvf: Remove deprecated messages

Need to remove deprecated messages:
ACTION_CONTROL_CLIENT_STATISTICS_11K_REQUEST
ACTION_CONTROL_CLIENT_STATISTICS_11K_RESPONSE

https://jira.prplfoundation.org/browse/PPM-291

Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

Commit:
agent: son_slave: Remove case from VS message handler

Need to remove case from VS message
handler for processing message:
ACTION_CONTROL_CLIENT_STATISTICS_11K_REQUEST

https://jira.prplfoundation.org/browse/PPM-291

Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

Commit:
controller: son_master: Remove case from VS message handler

Need to remove case for processing
ACTION_CONTROL_CLIENT_STATISTICS_11K_RESPONSE
message from VS message handler.

https://jira.prplfoundation.org/browse/PPM-291

Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

Commit:
controller: son_management: Remove deprecated case from handler

Need to remove deprecated case
for processing message from
handler:
ACTION_CLI_CLIENT_STATISTICS_11K_REQUEST

https://jira.prplfoundation.org/browse/PPM-291

Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

Commit:
controller: beerocks_cli: Remove deprecated CLI command

Need to remove deprecated CLI
command client_statistics_11k_req
and all related code.

https://jira.prplfoundation.org/browse/PPM-291

Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

Commit:
agent: son_slave: Remove case for deprecated message

Need to remove deprecated case from handler
for vendor specific message
ACTION_MONITOR_CLIENT_STATISTICS_11K_RESPONSE

https://jira.prplfoundation.org/browse/PPM-291

Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

Commit:
common: tlvf: Remove deprecated VS message

Need to remove deprecated VS message
ACTION_MONITOR_CLIENT_STATISTICS_11K_RESPONSE.

https://jira.prplfoundation.org/browse/PPM-291

Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

kantera800 commented:
@VladyslavTupikin  please  remove also monitor related CMDUs 
ACTION_MONITOR_CLIENT_STATISTICS_11K_REQUEST 
...
and then you can merge

Commit:
agent: fronthaul_manager: Remove deprecated VS message

Need to remove code which uses VS message:
ACTION_MONITOR_CLIENT_STATISTICS_11K_REQUEST
ACTION_MONITOR_CLIENT_STATISTICS_11K_RESPONSE

https://jira.prplfoundation.org/browse/PPM-291

Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

Commit:
common: bwl: Remove RRM_STA_Statistics_Response

https://jira.prplfoundation.org/browse/PPM-291

Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

Commit:
common: bwl: Remove deprecated method

Need to remove deprecated method
sta_statistics_11k_request().

https://jira.prplfoundation.org/browse/PPM-291

Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

Commit:
agent: monitor_thread: Remove deprecated methods

Need to remove deprecated methods
debug_statistics_11k_request()
debug_statistics_11k_response()

https://jira.prplfoundation.org/browse/PPM-291

Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

mariomaz commented on common/beerocks/bwl/dummy/mon_wlan_hal_dummy.cpp:
```diff
@@ -143,12 +143,6 @@ bool mon_wlan_hal_dummy::sta_beacon_11k_request(const SBeaconRequest11k &req, in
     return true;
 }
 
-bool mon_wlan_hal_dummy::sta_statistics_11k_request(const SStatisticsRequest11k &req)
```
According to the [commits ](https://github.com/prplfoundation/prplMesh/blob/master/CONTRIBUTING.md#commits)section in CONTRIBUTING.md: "*Every commit must stand on its own. No commit should ever break something.*"

If you remove a method that is pure virtual in the base class and overridden in derived classes, you must remove them all at once. Put in other words, remove sta_statistics_11k_request from all dummy, dwpal and nl80211 libraries in the same commit. Otherwise the commit is not complete (code must compile without errors after every single commit).

VladyslavTupikin replied:
fixed

mariomaz commented on common/beerocks/bwl/include/bwl/mon_wlan_hal.h:
```diff
@@ -55,7 +55,6 @@ class mon_wlan_hal : public virtual base_wlan_hal {
 
     virtual bool sta_channel_load_11k_request(const SStaChannelLoadRequest11k &req)      = 0;
     virtual bool sta_beacon_11k_request(const SBeaconRequest11k &req, int &dialog_token) = 0;
-    virtual bool sta_statistics_11k_request(const SStatisticsRequest11k &req)            = 0;
```
Just squash this commit together with the three previous commits

VladyslavTupikin replied:
solved

mariomaz commented on agent/src/beerocks/slave/son_slave_thread.cpp:
```diff
@@ -2777,25 +2777,6 @@ bool slave_thread::handle_cmdu_monitor_message(Socket *sd,
         send_cmdu_to_controller(cmdu_tx);
         break;
     }
-    case beerocks_message::ACTION_MONITOR_CLIENT_STATISTICS_11K_RESPONSE: {
```
This commit should be placed *before* those that remove ACTION_MONITOR_CLIENT_STATISTICS_11K_RESPONSE from TLVF

VladyslavTupikin replied:
fixed

VladyslavTupikin replied:
solved

mariomaz reviewed (approved):
All right except that some commits must be reordered and others squashed

Commit:
common: tlvf: Remove deprecated CLI message

Need to remove deprecated CLI message
ACTION_CLI_CLIENT_STATISTICS_11K_REQUEST.

https://jira.prplfoundation.org/browse/PPM-291

Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>
